lib/hsa/hc_am.cpp

```diff
@@ -519,14 +519,13 @@ am_status_t am_map_to_peers(void* ptr, size_t num_peer, const hc::accelerator* p
         {
             // here, accelerator is the device, but used to query system memory pool
             ptrAcc = info._acc;
-            pool = static_cast<hsa_amd_memory_pool_t*>(ptrAcc.get_hsa_am_system_region()); 
+            pool = static_cast<hsa_amd_memory_pool_t*>(ptrAcc.get_hsa_am_system_region());
         }
         else
             return AM_ERROR_MISC;
     }
 
-    const size_t max_agent = hc::accelerator::get_all().size();
-    hsa_agent_t agents[max_agent];
+    std::vector<hsa_agent_t> agents{hc::accelerator::get_all().size()};
```

Comment:
@AlexVlx could you separate fixes in AM library implementation into a separate PR?